### PR TITLE
fix user sign up do not have company info

### PR DESCRIPTION
**Pull request by sweiyang**

### CODE REVIEW

**Comment by cherrythia** [comment]
@sweiyang , cognito has an option to add custom attributes. It can be an option to add company into the cognito instead of saaving to our own db. 

No right or wrong, what is your view? Wanna learn from you. 

image

File: handler/authentication_handler.go [hunk 1]
```diff
@@ -28,28 +35,65 @@ func LoginHandler(c *gin.Context) {
...
 
-	err := c.BindJSON(&credentials)
+	err := c.BindJSON(&signUpRequest)
 	if err != nil {
 		logrus.WithField("err", err).Error("error params")
 		c.JSON(http.StatusBadRequest, CreateResponse(fmt.Sprintf("%v", err)))
 		return
 	}
+	// check if company name already exist
+	if provider.ProviderControllerObj.IsCompanyExist(signUpRequest.CompanyName) {
```
Comment (cherrythia) [inline handler/authentication_handler.go:65]: @sweiyang , this scenario means each provider user requires a unique company. You cannot have 2 employees of the same company to sign up.

File: handler/authentication_handler.go [hunk 2]
```diff
@@ -28,28 +35,65 @@ func LoginHandler(c *gin.Context) {
...
 		return
 	}
+	// get compnay name
+	providerDetails, err := provider.ProviderControllerObj.GetProvider(credentials.Email)
+	if err != nil {
+		logrus.WithField("err", err).Error("error getting provider")
+		c.JSON(http.StatusBadRequest, CreateResponse(fmt.Sprintf("%v", err)))
+		return
+	}
+	resp.CompanyName = providerDetails.CompanyName
```
Comment (cherrythia) [inline handler/authentication_handler.go:45]: your approach will have companyName as a different parameter. 

image

File: handler/authentication_handler.go (hunk 1 above)
Comment (sweiyang) [reply handler/authentication_handler.go:65]: nope cannot. The first provider that signs up as a new company will be a super admin. Afterwards, we will have a sections for super admin to create employee account to access

File: handler/authentication_handler.go (hunk 2 above)
Comment (sweiyang) [reply handler/authentication_handler.go:45]: if needed might need a way to included within jwt

**Comment by sweiyang** [comment]
@cherrythia I didnt think of using cognito. I think we can try. need to see if able to get all user by company

**Pull request merged**